src/wallet/wallet.cpp

```diff
@@ -3973,7 +4023,8 @@ CWallet* CWallet::CreateWalletFromFile(const std::string& name, const fs::path&
 
     int64_t nStart = GetTimeMillis();
     bool fFirstRun = true;
-    CWallet *walletInstance = new CWallet(name, WalletDatabase::Create(path));
+    auto wallet_deleter = MakeUnique<CWallet>(name, WalletDatabase::Create(path));
```

Comment:
The name `wallet_deleter` scares me.

Reply:
is `temp_wallet` (with a comment above) less scary?

Reply:
Yes, certainly. What I'd prefer, ideally, is to just call this wallet_instance and use this smart pointer everywhere instead of `CWallet* walletInstance = temp_wallet.get()`.
But this is fine, it's no longer scary.

Reply:
ok, I'll change it if I need to update the branch for another reason.